### PR TITLE
MBS-10989 extract `:common:report-api` module

### PR DESCRIPTION
**Pull request by dsvoronin**
a bunch of changes around generic report models and classes to separate it from avito-specific

### CODE REVIEW

File: subprojects/common/report-api/src/main/kotlin/com/avito/report/ReportLinkGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+public interface ReportLinkGenerator {
+
+    public fun generateReportLink(
+        filterOnlyFailtures: Boolean = false,
+        team: String? = null
+    ): String
+
+    public fun generateTestLink(testName: TestName): String
+
+    public class NoOp(
```
Comment (sergioboi) [inline subprojects/common/report-api/src/main/kotlin/com/avito/report/ReportLinkGenerator.kt]: impl

File: subprojects/common/report-api/src/main/kotlin/com/avito/report/TestSuiteNameProvider.kt [hunk 2]
```diff
@@ -0,0 +1,11 @@
+package com.avito.report
+
+public interface TestSuiteNameProvider {
+
+    public fun getName(): String
+
+    public class NoOp(private val name: String = "Unknown") : TestSuiteNameProvider {
```
Comment (sergioboi) [inline subprojects/common/report-api/src/main/kotlin/com/avito/report/TestSuiteNameProvider.kt]: impl

File: subprojects/common/report-api/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+plugins {
```
Comment (sergioboi) [inline subprojects/common/report-api/build.gradle.kts:1]: this module looks non-independent
In module `report-api` I expect to find some interface `Report` with contract
And anyone could create an impl module for that `Report`

What I actually see: bunch of models I don't know how to use

So now module looks like a `report-models` module

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProvider.kt [hunk 4]
```diff
@@ -14,17 +14,17 @@ import java.io.File
  *
  * use [TestArtifactsProviderFactory] to create instances
  */
-interface TestArtifactsProvider {
+public interface TestArtifactsProvider {
```
Comment (sergioboi) [inline subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProvider.kt:17]: As I see this interface doesn't relate to `report-viwer`

Comment (dsvoronin) [reply subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProvider.kt:17]: yes, seems it should be another module

**Review by sergioboi — APPROVED** [review]

File: subprojects/common/report-api/build.gradle.kts (hunk 3 above)
Comment (dsvoronin) [reply subprojects/common/report-api/build.gradle.kts:1]: Sure, just not in single PR;
There are a lot of weird dependencies i am trying to unravel right now )

File: subprojects/common/report-api/src/main/kotlin/com/avito/report/ReportLinkGenerator.kt (hunk 1 above)
Comment (dsvoronin) [reply subprojects/common/report-api/src/main/kotlin/com/avito/report/ReportLinkGenerator.kt]: extracted to separate class with more meaningful name

File: subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProvider.kt (hunk 4 above)
Comment (dsvoronin) [reply subprojects/common/report-viewer/src/main/kotlin/com/avito/report/TestArtifactsProvider.kt:17]: done

**Pull request merged**